src/shapes/Object/Object.ts

```diff
@@ -70,6 +71,34 @@ export type TCachedFabricObject<T extends FabricObject = FabricObject> = T &
     _cacheContext: CanvasRenderingContext2D;
   };
 
+export type ObjectToCanvasElementOptions = {
+  format?: ImageFormat;
+  // Multiplier to scale by
+  multiplier?: number;
+  // Cropping left offset. Introduced in v1.2.14
+  left?: number;
+  // Cropping top offset. Introduced in v1.2.14
+  top?: number;
+  // Cropping width. Introduced in v1.2.14
+  width?: number;
+  // Cropping height. Introduced in v1.2.14
+  height?: number;
+  // Enable retina scaling for clone image. Introduce in 1.6.4
+  enableRetinaScaling?: boolean;
+  // Remove current object transform ( no scale , no angle, no flip, no skew ). Introduced in 2.3.4
+  withoutTransform?: boolean;
+  // Remove current object shadow. Introduced in 2.4.2
+  withoutShadow?: boolean;
+  // Account for canvas viewport transform
+  viewportTransform?: boolean;
+  // Function to create the output canvas to export ont
+  canvasProvider?: (el?: HTMLCanvasElement) => Canvas;
```

Comment:
```suggestion
  canvasProvider?: (el?: HTMLCanvasElement) => StaticCanvas;
```

Reply:
Or maybe we should pass a generic here `<T extends StaticCanvas>`

Reply:
yes maybe a generic is better. i ll see if i can make it without triggerin 1000 warning. If i trigger 1000 warning i l leave it as StaticCanvas

Reply:
It might because of a def cycle

Reply:
Maybe better to define the interface needed for the method